core/src/main/java/hudson/util/ProcessTree.java

```diff
@@ -498,6 +520,7 @@ public void killRecursively() throws InterruptedException {
                 return;
 
             LOGGER.log(FINER, "Killing recursively {0}", getPid());
+            killSoftly(getPid());
```

Comment:
We'll send Ctrl+C only to the root of a process tree. That's alright for my use case.
In case every process in the tree should receive Ctrl+C, we should change the implementation of WinProcess and do Ctrl+C sending in killRecursively and kill (should then pass a timeout to it).